GVFS/GVFS.Platform.Mac/GVFS.Platform.Mac.csproj

```diff
@@ -4,6 +4,7 @@
     <TargetFrameworks>netcoreapp2.1;netstandard2.0</TargetFrameworks>
     <Platforms>x64</Platforms>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>

```

Comment:
What runtime were we targeting before this change?

Or (based on https://aka.ms/dotnet-runtime-patch-selection) were we always publishing with the latest but building with a version that could potentially be different?

Reply:
I think the problem is that the IDE tooling did not always specify a run-time version during the publish step, while our build scripts do. I *suspect* this lack of an explicit runtime identifier in the IDE builds for publish is what is causing confusion... without the `TargetLatestRuntimePath` property, the IDE generates:

1) A non-self contained version that targets systems with a minimum Framework  version of 2.1
2) A self contained exe with a v 2.1.9 version of the runtime

I think the error is in case this discrepancy is not OK (i.e. you have a hard dependency on the 2.1.9 version of the runtime).

> were we always publishing with the latest but building with a version that could potentially be different?

No - I think this error is telling us that there would be a discrepancy. The build scripts include the runtime version in the publish step, so I think it could use that information when running the publish command, and not have ambiguity between the version targeted by a self contained exe versus a version that is meant to be run in environments where the system provides the Framework libs.

Reply:
> What runtime were we targeting before this change?

I think our build script would always generate self contained executables that targeted the latest patch version on the build machine.

Reply:
>I think the problem is that the IDE tooling did not always specify a run-time version during the publish step, while our build scripts do.

For my own understanding, does "run-time version" mean something like `--runtime osx-x64`?

> A self contained exe with a v 2.1.9 version of the runtime

Do you know where it was picking up 2.1.9 from (and how did you figure this out)?

Reply:
I just saw we were already including `<TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>` in most of the projects already (and that's probably reason enough to do it here).

Based on the link I think this sounds like us?

> Prefer the self-contained behavior. If the application is self-contained, this is the default behavior. If the application is framework-dependent, and requires the latest patch installed, set TargetLatestRuntimePatch to true in the project file.

"requires the latest patch installed" is the one part I'm not sure about.  Do we require that the latest patch is installed (it sounds like a good thing even if we don't have a strict dependency on it?).

This PR looks good, but I still don't have a good understanding of exactly what we're specifying when building and publishing.

CC: @nickgra as well, I saw he did some research into this in an earlier PR (https://github.com/microsoft/VFSForGit/pull/496#discussion_r232555856).

Reply:
> For my own understanding, does "run-time version" mean something like --runtime osx-x64?

I think when you build a self contained executable (I.e. You specify a runtime id in the former command, or the project file has a runtime specified), it chooses the latest runtime available by default.

The specific versions here came from the error messages I encountered

Reply:
> "requires the latest patch installed" is the one part I'm not sure about

I think this refers to "Framework Dependent Deployments" (FDD), or deployments where we would expect the target system to provide the framework. With this property set, "Self Contained Deployments" (which is what I think actually care about) include 2.1.9, and a FDD would require  the target system to have at 2.1.9 installed. Without this property, I think the publish command is complaining that FDD have a minimum target version of 2.1, but was built against the 2.1.9 runtime.

Even if we don't care about FDD, I suspect the publish commands issued by IDE builds still generates a FDD package (whereas our command line builds generally provide the '--runtime' parameter in the dotnet buil and publish commands) and this is causing ambiguity.

Reply:
> Even if we don't care about FDD, I suspect the publish commands issued by IDE builds still generates a FDD package (whereas our command line builds generally provide the '--runtime' parameter in the dotnet buil and publish commands) and this is causing ambiguity.

I believe even command line builds create the FDD assemblies first, then do the self-contained ones after? At least that's what I've seen.